### PR TITLE
Process 'Rehost'

### PR DESCRIPTION
**Pull request by micheljung**
Let the client process 'Rehost' messages and request a rehost from the server.

### CODE REVIEW

File: src/fa/game_session.py [hunk 1]
```diff
@@ -147,6 +154,9 @@ def _on_game_message(self, command, args):
             elif args[0] == 'Lobby':
                 # TODO: Eagerly initialize the game by hosting/joining early
                 pass
+        elif command == 'Rehost':
+            self._rehost = True
```
Comment (Sheeo) [inline src/fa/game_session.py]: This appears to be a write only property?

File: src/fa/game_session.py [hunk 2]
```diff
@@ -161,3 +171,19 @@ def _exited(self, status):
...
+        if self._rehost:
+            self.client.host_game(title=self.game_name,
+                                  mod=self.game_mod,
+                                  visibility=self.game_visibility,
+                                  mapname=self.game_map,
+                                  password=self.game_password,
+                                  is_rehost=True)
+
+        self._rehost = False
+        self.game_uid = None
```
Comment (Sheeo) [inline src/fa/game_session.py]: How about we store these in a dict instead?

**Comment by Sheeo** [comment]
I'll make the suggested amendments when merging this, sorry for the delay

File: src/fa/game_session.py (hunk 1 above)
Comment (micheljung) [inline src/fa/game_session.py]: No, see line 175; `if self._rehost`

File: src/fa/game_session.py (hunk 2 above)
Comment (micheljung) [inline src/fa/game_session.py]: I guess a dict would be ok, I'm just not a fan of losing the last bit of of compile time checks and auto completion one can get with python x) (it's easy to misspell a dict-key and you have to manually check how the one you need to set is named)

**Comment by micheljung** [comment]
Sure, no problem. I know you have other priorities busy atm.

**Comment by micheljung** [comment]
Related to https://github.com/FAForever/fa/pull/1330

**Comment by micheljung** [comment]
This shouldn't have been merged; the suggested amendments were never made, it was never tested and it doesn't even work.

**Pull request merged**